### ISSUE TITLE
Stack Trace in Async Zod Function

### ISSUE THREAD

**Issue by rubber-duck-software** [colinhacks/zod#572]
Currently, async zod functions do not produce a useful stack trace when their validation fails. The trace generally contains only zod internals, looking something like this... 

```
Error: [
  {
    "code": "invalid_arguments",
    "argumentsError": {
      "issues": [
        {
          "code": "too_small",
          "minimum": 1,
          "inclusive": true,
          "type": "array",
          "path": [],
          "message": "Should have at least 1 items"
        }
      ]
    },
    "path": [],
    "message": "Invalid function arguments"
  }
]
    at new ZodError (/Users/USERNAME/PROJECT/node_modules/zod/lib/ZodError.js:75:28)
    at handleArgs (/Users/USERNAME/PROJECT/node_modules/zod/lib/types.js:1786:23)
    at ZodFunction.<anonymous> (/Users/USERNAME/PROJECT/node_modules/zod/lib/types.js:1817:49)
    at step (/Users/USERNAME/PROJECT/node_modules/zod/lib/types.js:59:23)
    at Object.next (/Users/USERNAME/PROJECT/node_modules/zod/lib/types.js:40:53)
    at fulfilled (/Users/USERNAME/PROJECT/node_modules/zod/lib/types.js:31:58)
```

Async function validation should be reworked slightly to produce a more usable stack trace

**Comment by rubber-duck-software** [comment]
@colinhacks - could you take a look at this one? I think it resolves something you attempted in 3.7

**Comment by mvasin** [comment]
Hi! I'm on version 3.20.2, and the problem persists. Here's the stack trace I receive:

```
/Users/user/project/node_modules/zod/lib/types.js:29
        const error = new ZodError_1.ZodError(ctx.common.issues);
                      ^
ZodError: [
  {
    "code": "invalid_type",
    "expected": "string",
    "received": "undefined",
    "path": [],
    "message": "Required"
  }
]
    at handleResult (/Users/user/project/node_modules/zod/lib/types.js:29:23)
    at ZodString.safeParse (/Users/user/project/node_modules/zod/lib/types.js:142:16)
    at ZodString.parse (/Users/user/project/node_modules/zod/lib/types.js:122:29)
    at Object.<anonymous> (/Users/user/project/src/config.ts:80:28)
    at Module._compile (node:internal/modules/cjs/loader:1159:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1213:10)
    at Module.load (node:internal/modules/cjs/loader:1037:32)
    at Function.Module._load (node:internal/modules/cjs/loader:878:12)
    at Function.mod._load (/Applications/HTTP Toolkit.app/Contents/Resources/httptoolkit-server/overrides/js/wrap-require.js:53:33)
    at Module.require (node:internal/modules/cjs/loader:1061:19)
```
It still gives no clue where _in my app_ did the problem occur.
